rust/src/writer/utils.rs

```diff
@@ -235,3 +282,113 @@ impl Write for ShareableBuffer {
         (*inner).flush()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::array::{
+        BinaryArray, BooleanArray, Date32Array, Date64Array, Int16Array, Int32Array, Int64Array,
+        Int8Array, LargeBinaryArray, StringArray, TimestampMicrosecondArray,
+        TimestampMillisecondArray, TimestampNanosecondArray, TimestampSecondArray, UInt16Array,
+        UInt32Array, UInt64Array, UInt8Array,
+    };
+
+    #[test]
+    fn test_stringified_partition_value() {
+        let reference_pairs: Vec<(Arc<dyn Array>, Option<String>)> = vec![
+            (Arc::new(Int8Array::from(vec![None])), None),
+            (
+                Arc::new(Int8Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(Int16Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(Int32Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(Int64Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt8Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt16Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt32Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt64Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(StringArray::from(vec![
+                    Some(String::from("1")),
+                    Some(String::from("2")),
+                ])),
+                Some(String::from("1")),
+            ),
```

Comment:
The `StringArray::from` implementation can take a `&str`:

```suggestion
            (Arc::new(StringArray::from(vec!["1"])), Some("1")),
```